raiden_libs/network/matrix/client.py

```diff
@@ -22,9 +23,38 @@ class GMatrixHttpApi(MatrixHttpApi):
     of concurrent requests we make to the number of connections
     available to us in requests.Session connection pool size.
     """
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.lock = Semaphore(DEFAULT_POOLSIZE)
+    def __init__(
+            self,
+            base_url,
+            token=None,
+            identity=None,
+            default_429_wait_ms=5000,
+            max_retries=3,
+            pool_maxsize=256,
+    ):
+        super().__init__(
+            base_url=base_url,
+            token=token,
+            identity=identity,
+            default_429_wait_ms=default_429_wait_ms,
+        )
+
+        session = Session()
```

Comment:
The session is already assigned [here](https://github.com/matrix-org/matrix-python-sdk/blob/50676efbb36a010d13b9901413a39d84fdb0d89e/matrix_client/api.py#L57), no need to create a new one, you may just `mount` the `HTTPAdapter`s on it.